### DISCUSSION TITLE
How do I add tags to entries on the blog start page? My links aren't picking up the URL base there, but do elsewhere.

### DISCUSSION THREAD

**Discussion by cellio** [datenstrom/community#698] [Ask a question] [answered]
I have seen [this discussion from 2019](https://github.com/datenstrom/yellow/discussions/393), but the answer there refers to files that don't exist in today's distribution so I assume something's been refactored since then.

My tag links are working correctly in the following locations:
- in the sidebar
- on the pages for individual entries

I would like to also show tag links with each entry on main blog pages.  Taking inspiration from the "author" layout and individual blog posts (`system/layouts/blog.html`), I added the following to `system/layouts/blog-start.html` after entry-meta, before entry-content (line breaks added here for legibility):

    <div class="entry-tags">
    <p><?php echo $this->yellow->language->getTextHtml("blogTag") ?> 
    <?php $tagCounter = 0; foreach (preg_split("/\s*,\s*/", $page->get("tag")) as $tag) { if (++$tagCounter>1) echo ", "; 
    echo "<a href=\"".$this->yellow->page->getPage("blogStart")->getLocation(true).$this->yellow>toolbox->normaliseArguments("tag:$tag")."\">".htmlspecialchars($tag)."</a>"; } ?></p>
    </div>

This does not seem to be picking up the location prefix.  Instead of URLs like `example.com/blog/tag:something/`, which work, I'm getting URLs like `tag:something/`, which don't.

I realize that on an individual blog page "this" is an entry while on blog-start "this" is the "container" (the blog).  But I thought that `$this->yellow` would reach the same destination from any page in the site, and so `$this->yellow->page->getPage("blogStart")->getLocation(true)` would behave the same from both starting points.  I suspect that that's not actually a correct assumption, but as a PHP newbie I'm not sure where to go from here.  Can anyone help me figure out what I'm doing wrong?

**Comment by schulle4u** [comment, marked as answer]
I think you're almost there. In the blog-start layout you should use `getLocation(true)` without additional API calls. Have a look at the author URL code in the entry meta section, it slightly differs from your tag URL's. Here is an example to show tags in blog-start layout. 

```
<?php if ($page->isExisting("tag")): ?>
<div class="entry-tags">
<p><?php echo $this->yellow->language->getTextHtml("blogTag") ?> <?php $tagCounter = 0; foreach (preg_split("/\s*,\s*/", $page->get("tag")) as $tag) { if (++$tagCounter>1) echo ", "; echo "<a href=\"".$this->yellow->page->getLocation(true).$this->yellow->toolbox->normaliseArguments("tag:$tag")."\">".htmlspecialchars($tag)."</a>"; } ?></p>
</div>
<?php endif ?>
```

Hope this helps as a starting point. 

Steffen

**Reply by cellio** [reply]
Thank you!  That did it, and I now see what I was doing wrong with that extra call.

**Reply by markseuffert** [reply]
I edited the layout name to make this answer easier to find. Hope that's okay with you.
